survey/src/test/java/com/formulai/survey/integrationTests/SurveyControllerTest.java

```diff
@@ -0,0 +1,170 @@
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.*;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.status;
+
+@SpringBootTest
+@AutoConfigureMockMvc
+@Transactional
+class SurveyControllerTest {
+
+    @Autowired
+    private MockMvc mockMvc;
+
+    @Autowired
+    private SurveyRepository surveyRepository;
+
+    @Autowired
+    private ObjectMapper objectMapper;
+
+    @Test
+    void getSurvey_shouldReturnSurveyById() throws Exception {
+        // given
+        Survey firstSurvey = surveyRepository.findAll().stream()
+                .findFirst()
+                .orElseThrow(() -> new RuntimeException("No surveys found"));
+        UUID surveyId = firstSurvey.getId();
+
+        // when
+        MvcResult result = mockMvc.perform(get("/v1/surveys/" + surveyId))
+                .andExpect(status().isOk())
+                .andReturn();
+
+        // then
+        SurveyResponse response = objectMapper.readValue(
+                result.getResponse().getContentAsString(), SurveyResponse.class);
+
+        assertNotNull(response);
+        assertEquals(firstSurvey.getId(), response.id());
+        assertEquals(firstSurvey.getName(), response.name());
+        assertEquals(firstSurvey.getSchemaJson(), response.schemaJson());
+    }
+
+    @Test
+    void getAllSurveys_shouldReturnAllSurveys() throws Exception {
+        // given
+        List<Survey> surveys = surveyRepository.findAll();
+
+        // when
+        MvcResult result = mockMvc.perform(get("/v1/surveys"))
+                .andExpect(status().isOk())
+                .andReturn();
+
+        // then
+        List<SurveyResponse> responses = objectMapper.readValue(
+                result.getResponse().getContentAsString(),
+                objectMapper.getTypeFactory().constructCollectionType(List.class, SurveyResponse.class));
+
+        assertNotNull(responses);
+        assertEquals(surveys.size(), responses.size());
+    }
+
+    @Test
+    void getAllSurveyResponses_shouldReturnAllResponsesForSurvey() throws Exception {
+        // given
+        Survey firstSurvey = surveyRepository.findByName("example")
+                .orElseThrow(() -> new RuntimeException("No surveys found"));
+        UUID surveyId = firstSurvey.getId();
+
+        // when
+        MvcResult result = mockMvc.perform(get("/v1/surveys/" + surveyId + "/answers"))
+                .andExpect(status().isOk())
+                .andReturn();
+
+        // then
+        List<SurveyAnswerResponse> responses = objectMapper.readValue(
+                result.getResponse().getContentAsString(),
+                objectMapper.getTypeFactory().constructCollectionType(List.class, SurveyAnswerResponse.class));
+
+        assertNotNull(responses);
+        assertFalse(responses.isEmpty());
+        assertEquals(surveyId, responses.getFirst().surveyId());
+    }
+
+
+    @Test
+    void createSurvey_shouldReturn400WithInvalidRequest() throws Exception {
```

Comment:
Nit: The test name doesn't really tell us what's the specific case that's being tested here. How is the request invalid? 
In test below you specify both, the outcome and a reason `createSurvey_shouldReturn400WithEmptyName`